rotkehlchen/chain/arbitrum_one/modules/umami/balances.py

```diff
@@ -0,0 +1,221 @@
+                        token_decimals=vault_token.decimals,
+                    )
+                    balances[user_address].assets[vault_token] += Balance(
+                        amount=amount,
+                        usd_value=amount * formatted_price_per_share * underlying_token_price,
+                    )
+            except RemoteError as e:
+                log.error(f'Failed to query umami balances for vault {vault_token.evm_address}: {e!s}')  # noqa: E501
+                continue
+
+        # get staked balances.
+        for user_address in addresses_with_deposits:
+            try:
+                results = self.evm_inquirer.multicall(
+                    calls=[
+                        (
+                           self.umami_masterchef_contract.address,
+                           self.umami_masterchef_contract.encode(
+                               method_name='userInfo',
+                               arguments=[idx, user_address],
+                           ),
+                        ) for idx in range(4)
+                    ],
+                    call_order=call_order,
+                    calls_chunk_size=3,  # etherscan throws an error for >3 chunk size.
+                )
+                if not results:
+                    log.error(f'Failed to query umami vault balances for address {user_address}')
+                    continue
+
+                for idx, result in enumerate(results):
+                    balance, _ = self.umami_masterchef_contract.decode(
+                        result=result,
+                        method_name='userInfo',
+                        arguments=[idx, user_address],
+                    )
+
+                    if balance == 0:
+                        continue
+
+                    gm_vault_token_address = self.get_gm_vault_address_by_index(idx)
+                    gm_vault_token = GlobalDBHandler().get_evm_token(
+                        address=gm_vault_token_address,
+                        chain_id=self.evm_inquirer.chain_id,
+                    )
+                    assert gm_vault_token is not None, f'vault token {gm_vault_token_address} must be present'  # noqa: E501
+
+                    underlying_vault_token = gm_vault_token.underlying_tokens[0]
+                    underlying_token_price = Inquirer().find_usd_price(Asset(underlying_vault_token.get_identifier(self.evm_inquirer.chain_id)))  # noqa: E501
+                    amount = token_normalized_value_decimals(
+                        token_amount=balance,
+                        token_decimals=gm_vault_token.decimals,
+                    )
+                    price_per_share = self.evm_inquirer.call_contract(
+                        contract_address=gm_vault_token.evm_address,
+                        abi=UMAMI_ASSET_VAULT_ABI,
+                        method_name='pps',
+                    )
+                    formatted_price_per_share = token_normalized_value_decimals(
```

Comment:
the code here is very similar to the code above. Abstract them into a common fucntion and use them once for staking and once for deposit balances.